### PR TITLE
clickhouse: divided into lts package and stable

### PR DESCRIPTION
**Pull request by patryk4815**
###### Description of changes

Hi. Currently there is only available "lts" package and it is not possible to install "stable". 
So i divided it into lts package and stable :) 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by patryk4815** [comment]
@ofborg build clickhouse clickhouse-lts

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23520,12 +23520,12 @@ with pkgs;
...
 
-  clickhouse = callPackage ../servers/clickhouse {
-    # upstream requires llvm12 as of v22.3.2.2
+  inherit (callPackage ../servers/clickhouse {
+    # upstream requires llvm13 as of v22.8.5.29
     inherit (llvmPackages_13) clang-unwrapped lld llvm;
     llvm-bintools = llvmPackages_13.bintools;
     stdenv = llvmPackages_13.stdenv;
-  };
+  }) clickhouse clickhouse-lts;
```
Comment (cpcloud) [inline pkgs/top-level/all-packages.nix:23528]: The pattern I've seen elsewhere in this file is to define a core `.nix` file with input parameters like `version_suffix` (which should probably be renamed to `version-suffix`), and then invoke that twice to get two separate packages, like this:

```nix
clickhouse = callPackage ../servers/clickhouse/core.nix {
  version-suffix = "stable";
  ...
};

clickhouse-lts = callPackage ../servers/clickhouse/core.nix {
  version-suffix = "lts";
  ...
};
```

Comment (patryk4815) [reply pkgs/top-level/all-packages.nix:23528]: I'm not sure, then `hash` should be put in `all-packages.nix`. But I see no one puts hash there hmmm

Comment (patryk4815) [reply pkgs/top-level/all-packages.nix:23528]: I took this example from: https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/hbase/default.nix

**Comment by patryk4815** [comment]
@ofborg build clickhouse clickhouse-lts

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:23528]: We can do that. Repeating the path twice is not that nice.

File: pkgs/servers/clickhouse/default.nix [hunk 2]
```diff
@@ -6,71 +6,86 @@
...
+      icu jemalloc libxml2 lld llvm lz4 libmysqlclient openssl perl
+      poco protobuf python3 rapidjson re2 rdkafka readline sparsehash unixODBC
+      xxHash zstd
+    ] ++ lib.optional stdenv.hostPlatform.isx86 libcpuid;
 
-  postPatch = ''
-    patchShebangs src/
+    postPatch = ''
+      patchShebangs src/
+      patchShebangs cmake/ld.lld.in
```
Comment (SuperSandro2000) [inline pkgs/servers/clickhouse/default.nix:34]:
```suggestion
      patchShebangs src/ cmake/ld.lld.in
```

File: pkgs/servers/clickhouse/default.nix [hunk 3]
```diff
@@ -6,71 +6,86 @@
 , nixosTests
 }:
 
-stdenv.mkDerivation rec {
-  pname = "clickhouse";
-  version = "22.8.5.29";
+let
+  common = {version-type, version, hash}: stdenv.mkDerivation {
+    pname = "clickhouse-${version-type}";
```
Comment (SuperSandro2000) [inline pkgs/servers/clickhouse/default.nix:11]:
```suggestion
    pname = "clickhouse";
```

File: pkgs/servers/clickhouse/default.nix [hunk 4]
```diff
@@ -6,71 +6,86 @@
 , nixosTests
 }:
 
-stdenv.mkDerivation rec {
-  pname = "clickhouse";
-  version = "22.8.5.29";
+let
+  common = {version-type, version, hash}: stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/servers/clickhouse/default.nix:10]:
```suggestion
  common = { version, hash }: stdenv.mkDerivation {
```

File: pkgs/servers/clickhouse/default.nix [hunk 5]
```diff
@@ -6,71 +6,86 @@
...
+  };
+in
+{
+  clickhouse = common {
+    version-type = "stable";
+    version = "22.10.2.11";
+    hash = "sha256-jUS5EMrr5NPQld4I8d10AlIhQFaNLi6Q+VczDAd7P2U=";
+  };
+  clickhouse-lts = common {
+    version-type = "lts";
```
Comment (SuperSandro2000) [inline pkgs/servers/clickhouse/default.nix:87]:
```suggestion
    version = "22.10.2.11";
    hash = "sha256-jUS5EMrr5NPQld4I8d10AlIhQFaNLi6Q+VczDAd7P2U=";
  };
  clickhouse-lts = common {
```

**Comment by SuperSandro2000** [comment]
> Hi. Currently there is only available "lts" package and it is not possible to install "stable".
> So i divided it into lts package and stable :)

TBH do we really need lts? NixOS is pretty much the opposite of LTS and IMO stable is enough.

**Comment by patryk4815** [comment]
@ofborg build nixStatic

**Pull request closed without merging**